### PR TITLE
Merge temporary assembly Expressions.Properties into Expressions

### PR DESCRIPTION
**Pull request by tomlm**
* Switch to string Interpolation instead of LGFile
* Cleaned up semantics around switchcondition (it's not a stringExpression) and case (it's always a constant). 
* Moved Evaluate() from extension method to LGFile

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by chrimc62** [comment]
> using Newtonsoft.Json;

Did you decide to keep this namespace?

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs:11 in bac6832. [](commit_id = bac6832885a725b7ebe494347a59727e1b18cbc5, deletion_comment = False)

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public SwitchCondition([CallerFilePath] string callerPath = "", [CallerLineNumbe
         /// Value Expression against memory. This value expression will be combined with value expression in case statements to make a bool expression.
         /// </value>
         [JsonProperty("condition")]
-        public ValueExpression Condition { get; set; } 
+        public Expression Condition { get; set; }
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs]: >Expression [](start = 15, length = 10)

You want to require an expression?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs [hunk 2]
```diff
@@ -121,8 +121,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
                         foreach (var cse in this.Cases)
                         {
-                            // Values for cases are always coerced to string
-                            var caseCondition = Expression.EqualsExpression(this.Condition.ToExpression(), cse.CreateValueExpression());
+                            Expression caseExpression = null;
+
+                            // Values for cases are always coerced to constant
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs]: >// Values for cases are always coerced to constant [](start = 28, length = 50)

Any particular reason you want to force these to be constants?  We don't enforce that for if and there is no particular efficiency concern with giving these the opportunity to be computed.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.SwitchCondition.schema [hunk 3]
```diff
@@ -49,8 +48,9 @@
                         "title": "Value",
                         "description": "Value.",
                         "examples": [
-                            "'red'",
-                            "dialog.colors.red"
+                            "red",
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.SwitchCondition.schema]: >  "red", [](start = 26, length = 8)

Given your changes wouldn't this have to be in quotes?

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFile.cs [hunk 4]
```diff
@@ -132,6 +132,37 @@ public object EvaluateTemplate(string templateName, object scope = null)
             return evaluator.EvaluateTemplate(templateName, scope);
         }
 
+        /// <summary>
+        /// Use to evaluate an inline template str.
+        /// </summary>
+        /// <param name="inlineStr">inline string which will be evaluated.</param>
+        /// <param name="scope">scope object or JToken.</param>
+        /// <returns>Evaluate result.</returns>
+        public object Evaluate(string inlineStr, object scope = null)
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFile.cs]: >Evaluate [](start = 22, length = 8)

Why is this in this change?  How does it relate to the lg changes for string interpolation?

File: tests/Microsoft.Bot.Builder.TestBot.Json/Samples/GeneratedForm/luis/luis.settings.chrimc.westus.json [hunk 5]
```diff
@@ -1,5 +0,0 @@
-{
-    "luis": {
-        "sandwich_en-us_lu": "236ff454-e019-4ab3-ad5f-0ae22d65ad33"
-    }
-}
```
Comment (chrimc62) [inline tests/Microsoft.Bot.Builder.TestBot.Json/Samples/GeneratedForm/luis/luis.settings.chrimc.westus.json]: Why do you delete these?  What is our policy around these?  We need at least one checked-in for the mock stuff to work, but we also need people to be able to update the mock.  Whose settings win?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs (hunk 1 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs]: yes

Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs]: Yes, it doesn't make any sense to say 
switch (13)
or 
switch("Hello")

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs (hunk 2 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs]: This is just refactoring existing behavior.

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFile.cs (hunk 4 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFile.cs]: This was code which was in properties, but should be in LG

**Pull request merged**